lib/frontend_helpers/form_builder_helpers/dynamic_fields.rb

```diff
@@ -0,0 +1,139 @@
+# frozen_string_literal: true
+
+module FrontendHelpers
+  module FormBuilderHelpers
+    module DynamicFields
+      # Field helper to add associated records in a form, it generates a label and a link
+      # to dynamically add additional associated records.
+      #
+      #   <%= form_for @product do |f| %>
+      #     <%= f.dynamic_fields_group :items,
+      #         label: 'Product Items', button_text: 'Add product item' %>
+      #     <%= f.submit %>
+      #   <% end %>
+      #
+      # The label and button HTML can be customized by passing a block:
+      #
+      #   <%= f.dynamic_fields_group :items do %>
+      #     <label>Items</label>
+      #     <%= f.link_to_add_fields "Add Item", :items %>
+      #   <% end %>
+      #
+      # Requirements:
+      # - Partial for the associated fields with the name:
+      #     "_#{singluar_association_name}_fields.html.erb"
+      # - The template needs to have a top level element with the class:
+      #     "#{singluar_association_name}-fields"
```

Comment:
#     "_#{singular_association_name}_fields.html.erb"
      # - The template needs to have a top level element with the class:
      #     "#{singular_association_name}-fields"